dip-0027.md

```diff
@@ -0,0 +1,163 @@
+* [DIP-0007: LLMQ Signing Requests / Sessions](https://github.com/dashpay/dips/blob/master/dip-0007.md)
+* [DIP-0010: LLMQ InstantSend](https://github.com/dashpay/dips/blob/master/dip-0010.md)
+* [DIP-0011: Identities](https://github.com/dashpay/dips/blob/master/dip-0011.md)
+
+## System Requirements
+
+Two criteria must be met for the credit pool to operate effectively:
+
+1. The value of Dash in the credit pool must always correspond to the number of credits on Platform. The Core blockchain does this by tracking the amount of locked Dash and preventing Dash above that amount from being withdrawn.
+2. Withdrawing Dash from credits must be possible without requiring the Core chain to validate any Platform constructs, such as Identities. This minimizes the level of reliance between Core and Platform.
+
+## Overview
+
+When Dash is locked in the credit pool, the person who locked it can claim a corresponding number of credits for an Identity on Dash Platform. As the Identity performs actions on Platform, its balance is reduced by the cost of the actions while the masternode payout pool value is increased by the same amount. Over time, funds in the masternode payout pool are distributed to the Identities of masternodes that participated in Platform validator sets. The details of masternode Identities and Platform payouts will be the focus of future DIPs.
+
+Eventually, an Identity may want to withdraw their credits from Platform to regain access to the underlying locked Dash. Any Identity type can issue a withdrawal request to Dash Platform. If the Identity’s balance is sufficient, this will result in the formation of a transaction signed by a quorum and valid on the Core chain. This transaction unlocks the requested amount of Dash from the credit pool and sends it to the provided Dash address.
+
+## Asset Locking
+
+Dash Core participates in the credit creation process by locking Dash in the credit pool with an Asset Lock transaction.
+
+### Asset Lock Transaction
+
+The Asset Lock transaction (AssetLockTx) is a DIP-2-based special transaction that gives Platform the information required to assign credits to Identities. This special transaction is type 8.
+
+The format of the Asset Lock special transaction payload:
+
+| **Name** | **Type** | **Size**  | **Description** |
+| - | - | - | - |
+| version | uint8_t | 1 | Asset Lock transaction version number |
+| count | uint8_t | 1 | The number of accounts funded via this Asset Lock |
+| credit_outputs | vec\<TxOut> | 28 * count| The hash of the public key(s) used to claim credits on Dash Platform, as well as the amount of Dash in duffs transferred into credits for this public key hash |
+
+Each Asset Lock transaction must create an unspendable output by using an OP_RETURN as the first script. The following criteria apply to the OP_RETURN output:
+
+* It must not include any data
+* Its value is the amount of Dash to be locked
+* The sum of all TxOut amounts must equal the value of the output
```

Comment:
what does this mean?

Reply:
the special asset lock transaction can have multiple inputs and multiple outputs.

One of the outputs should be: OP_RETURN, do not include any data after OP_RETURN
Extra payload has array of TxOut and their sum should be equal to amount under OP_RETURN